service/history/mutableStateBuilder.go

```diff
@@ -1710,7 +1729,6 @@ func (e *mutableStateBuilder) ReplicateDecisionTaskStartedEvent(
 		di.Attempt = 0
 	}
 
-	e.executionInfo.State = persistence.WorkflowStateRunning
```

Comment:
there was a line setting workflow state to running in AddDecisionTaskScheduledEvent:
https://github.com/uber/cadence/pull/2059/files/51ee7c4b07cc5b72d0b1ff72f895f871949cd14b#diff-01045b0895719962d1f13440e8795d1bL1543

the new logic is in ReplicateDecisionTaskScheduledEvent:
https://github.com/uber/cadence/pull/2059/files/51ee7c4b07cc5b72d0b1ff72f895f871949cd14b#diff-01045b0895719962d1f13440e8795d1bR1632

since decision task schedule happens before decision task started, so this change should be fine